app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,76 @@
+class OrdersController < ApplicationController
+  before_action :find_user_orders, only: %i[index show]
+  before_action :find_order_items, only: :show
+  before_action :find_order_price, only: :show
+
+  def index
+    find_all_orders if current_user.admin?
+  end
+
+  def show; end
+
+  def find_user_orders
+    @orders = {}
+    order_carts = Cart.where('user_id = ? AND placed_order = ?', current_user.id, true)
+
+    order_carts.each do |cart|
+      current_order = Order.where('cart_id = ?', cart.id).last
+      user = User.find(Cart.find(current_order.cart_id).user_id)
+      insert_order_info(current_order, user)
+    end
+  end
+
+  def find_all_orders
+    order_carts = Cart.where('placed_order = ?', true)
+
+    order_carts.each do |cart|
+      current_order = Order.where('cart_id = ?', cart.id).last
+      user = User.find(Cart.find(current_order.cart_id).user_id)
+      insert_order_info(current_order, user)
+    end
+  end
+
+  def insert_order_info(order, user)
+    @orders[order.id] = {}
```

Comment:
I think here you could something like:

@orders[order.id] = {
  user_name: user.name,
  date: order.created_at,
  ...
}